src/network/networkd-link.c

```diff
@@ -2080,7 +2080,7 @@ static int link_down_handler(sd_netlink *rtnl, sd_netlink_message *m, Link *link
         return 1;
 }
 
-int link_down(Link *link) {
+int link_down(Link *link, link_netlink_message_handler_t callback) {
         _cleanup_(sd_netlink_message_unrefp) sd_netlink_message *req = NULL;
```

Comment:
But where is the use case we need another callback ?

Reply:
CAN uses that. It is changed when CAN related things are moved.